lib/internal/modules/esm/fetch_module.js

```diff
@@ -248,8 +246,9 @@ allowList.addRange('127.0.0.1', '127.255.255.255');
 async function isLocalAddress(hostname) {
   try {
     if (
-      StringPrototypeStartsWith(hostname, '[') &&
-      StringPrototypeEndsWith(hostname, ']')
+      hostname &&
```

Comment:
i think this skips that if it's not a string, it would have previously thrown - so i think you also need to add a check above the try `if (typeof hostname !== 'string') { return false; }`?

Reply:
I think this is unnecessary. Function expects that it receives a string, and this will be used by another Node.js function. Safe to assume that hostname is string. JSDoc agrees.

Reply:
Why not go with `hostname?.[0] === '['` and avoid the truthy check?

Reply:
`isLocalAddress` [receives](https://github.com/nodejs/node/blob/e2ba06c507aa0d8a28fd70b1f5468716844efb84/lib/internal/modules/esm/fetch_module.js#L273-L285) the hostname from a URL parsed object (and JSDoc also says it expects a string), isn't the hostname property of a URL instance always a string? And this function isn't exported, so that's why I kept it this way

Also, String.prototype methods don't throw when non-strings are passed I think, for me these just return false silently:

`String.prototype.startsWith(1, '')`
`String.prototype.startsWith(null, '')`

Reply:
indeed if we can be sure it's always a string then you don't even need the truthiness check or the `?.`

you're right tho, they don't throw, but they do coerce to a string :-)

Reply:
Oh! Didn't know they were coerced, cool!

Just for context: the truthiness checks were to prevent an empty string and so an out-of-bounds access on the string, which is discouraged by the V8 team as it does a proto chain lookup (string -> String -> Object)